clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -475,6 +492,7 @@ protected PutObjectResponseType doPutOperation(final PutObjectType request) thro
       } catch (Exception ex) {
         LOG.debug("Failed to fire reporting event for OSG object creation", ex);
       }
+      OSGUtil.setCorsInfo(request, response, bucket.getBucketUuid());
```

Comment:
There wasn't enough shared/common code between the bucket/object operation methods to put the CORS calls anywhere except in every method that implements a bucket or object operation. They're all there now.

Bucket UUID/name: Resolved, see above.